local-cli/install/install.js

```diff
@@ -10,7 +11,17 @@ log.heading = 'rnpm-install';
 function install(args, config) {
   const name = args[0];
 
-  var res = spawnSync('npm', ['install', name, '--save'], spawnOpts);
+  const projectDir = config.getProjectRoots()[0];
+  const isYarnAvailable =
+    yarn.getYarnVersionIfAvailable() &&
+    yarn.isGlobalCliUsingYarn(projectDir);
+
+  var res;
+  if (isYarnAvailable) {
+    res = spawnSync('yarn', ['add', name], spawnOpts);
+  } else {
+    res = spawnSync('npm', ['install', name, '--save'], spawnOpts);
```

Comment:
`manager.add('some-package')` // translates to `yarn add some-package` with yarn & `npm install some-package --save` with npm